core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/batch/BatchHandling.scala

```diff
@@ -112,7 +115,8 @@ trait BatchHandling extends LazyLogging with AttributeSupport {
             existingEntities.size != uniqueUpdateIdentifiers.size || (uniqueUpdateIdentifiers diff actualPointers).nonEmpty
           )
             throw new EntityNotFoundException(
-              s"expected ${uniqueUpdateIdentifiers.size} entities to be updated, but found ${existingEntities.size}"
+              s"Expected ${uniqueUpdateIdentifiers.size} entities to be updated, but found ${existingEntities.size}",
+              code = StatusCodes.BadRequest
```

Comment:
for batch upsert/update, we want to throw 400 if anything doesn't exist; this change is what necessitated the catch-and-rethrow for updating a single entity